### PR TITLE
jenkins-operator: use ProwJobID when looking for enqueued jobs

### PR DESCRIPTION
**Pull request by diversario**
ProwJobID is a unique identifier that doesn't depend on how the job was setup.

### CODE REVIEW

File: prow/jenkins/jenkins.go [hunk 1]
```diff
@@ -617,7 +622,7 @@ func (c *Client) BuildFromSpec(spec *prowapi.ProwJobSpec, buildID, prowJobID str
 
 // ListBuilds returns a list of all Jenkins builds for the
 // provided jobs (both scheduled and enqueued).
-func (c *Client) ListBuilds(jobs []string) (map[string]Build, error) {
+func (c *Client) ListBuilds(jobs []buildQueryParams) (map[string]Build, error) {
```
Comment (stevekuznetsov) [inline prow/jenkins/jenkins.go]: If this is an exported method of the client we need to export the type so it's possible to use the client outside of this package

**Comment by diversario** [comment]
@stevekuznetsov No, the test that could've been that one would be `TestListBuilds` but: the issue only affects jobs from the queue, and that test returns empty list for a Queue API call `{"items": []}`.

**Comment by stevekuznetsov** [comment]
Can we add such a test to ensure this continues to work as expected?

**Comment by diversario** [comment]
Test added.

**Comment by stevekuznetsov** [comment]
/lgtm
/approve

Sweet!

**Pull request merged**